prow/plank/controller.go

```diff
@@ -167,41 +160,42 @@ func (c *Controller) Sync() error {
 	close(pjCh)
 
 	errCh := make(chan error, len(pjs))
+	reportCh := make(chan kube.ProwJob, len(pjs))
```

Comment:
Are you worried that this channel going to require too much storage? I'm not sure I have enough familiarity to understand what problem there would be. If we want the channel buffer to be smaller we can also spawn this before the worker goroutines:

```go
reportErrCh := make(chan error, len(pjs))
go func() {
    for report := range reportCh {
        if err := c.report(report); err != nil {
            reportErrCh <- err
        }
    }
}()

// do mutliprocessing
close(reportErrCh)

var reportErrs []error
for reportErr := range reportErrCh {
    reportErrs = append(reportErrs, reportErr)
}